core/src/test/java/kafka/server/share/SharePartitionTest.java

```diff
@@ -3652,7 +3652,8 @@ public void testReleaseSingleRecordBatch() {
         assertEquals(0, sharePartition.nextFetchOffset());
         assertEquals(1, sharePartition.cachedState().size());
         assertEquals(RecordState.AVAILABLE, sharePartition.cachedState().get(0L).batchState());
-        assertEquals(1, sharePartition.cachedState().get(0L).batchDeliveryCount());
```

Comment:
Can we have check `assertEquals(1, sharePartition.cachedState().get(0L).batchDeliveryCount());` prio calling `releaseAcquiredRecords` which validates that deliveryCount was first incremented and then latter decremented?